src/test/java/Hardcore/page/GoogleCloudMainPage.java

```diff
@@ -0,0 +1,39 @@
+package Hardcore.page;
+
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+public class GoogleCloudMainPage extends AbstractPage {
+    String GC_BASE_URL = "https://cloud.google.com/";
+
+    @FindBy(name = "q")
+    WebElement searchInput;
+
+    @FindBy(xpath = "//div[@class='devsite-cse-results']")
+    WebElement resultsPage;
+
+    @FindBy(xpath = "//b[text()='Google Cloud Platform Pricing Calculator']")
+    WebElement resultPattern;
+
+    public GoogleCloudMainPage(WebDriver driver) {
+        super(driver);
+    }
+
+    public GoogleCloudMainPage openPage() {
+        driver.get(GC_BASE_URL);
+        return this;
+    }
+
+    public GoogleCloudMainPage searchTerm(String term) {
+        waitForWebElementVisible(searchInput).sendKeys(term + Keys.ENTER);
+        return this;
+    }
+
+    public GoogleCloudMainPage clickCalculatorPage() {
+        waitForWebElementVisible(resultsPage);
```

Comment:
I did it because without wait it wasn't obvious that we click on the link from results, since this step was passed too fast. I agree that this is dumb to have it like that in terms of the test itself. I did it just for the purposes of flow demonstration.